packages/backend/src/apps/formatter/actions/date-time/transforms/add-subtract-date-time/fields.ts

```diff
@@ -0,0 +1,80 @@
+import { type DurationLikeObject } from 'luxon'
+import z from 'zod'
+
+import { type TransformSpec } from '@/apps/formatter/common/transform-spec'
+import { ensureZodObjectKey } from '@/helpers/zod-utils'
+
+const opTypeEnum = z.enum(['add', 'subtract'])
+
+const timeUnitEnum = z.enum([
+  'seconds',
+  'minutes',
+  'hours',
+  'days',
+  'months',
+  // To simplify date math code, values have the same keys as luxton
+  // duration objects.
+] as const satisfies ReadonlyArray<keyof DurationLikeObject>)
+
+const opsSchema = z.object({
```

Comment:
comment: should the enum and schema go into a separate `schema.ts` file?

Reply:
i think is ok, I want to keep them in the same file for easy referencing